File: 01-ARRAYS/baseball-game.php
<?php

# https://leetcode.com/problems/baseball-game/
function calPoints(array $operations): int
{
    $stack = [];

    foreach ($operations as $i) {
        if ($i == '+') {
            $stack[] = array_sum(array_slice($stack, -2));
        } elseif ($i == 'D') {
            $stack[] = end($stack) * 2;
        } elseif ($i == 'C') {
             array_pop($stack);
        } else{
            $stack[] = $i;
        }
    }

    return array_sum($stack);
}

File: 01-ARRAYS/baseball-game.py
from typing import *


# https://leetcode.com/problems/baseball-game/
def calPoints(operations: List[str]) -> int:
    stack = []

    helpers = {
        '+': lambda: stack.append(stack[-1] + stack[-2]),
        'D': lambda: stack.append(stack[-1] * 2),
        'C': lambda: stack.pop(),
    }

    [helpers.get(i, lambda: stack.append(int(i)))() for i in operations]

    return sum(stack)

